sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/secret_client.hpp

```diff
@@ -235,6 +235,35 @@ namespace Azure { namespace Security { namespace KeyVault { namespace Secrets {
         std::string const& version,
         KeyvaultSecretProperties const& properties,
         Azure::Core::Context const& context = Azure::Core::Context()) const;
+
+    /**
+     * @brief Back up the specified secret.
+     * Requests that a backup of the specified secret be downloaded to the client.
+     * All versions of the secret will be downloaded.
+     * This operation requires the secrets/backup permission.
+     *
+     * @param name The name of the secret<span class="x x-first x-last">.</span>
+     * @param context The context for the operation can be used for request cancellation.
+     *
+     * @return The The backup blob containing the backed up secret.
+     */
+    Azure::Response<BackupSecretData> BackupSecret(
+        std::string const& name,
+        Azure::Core::Context const& context = Azure::Core::Context()) const;
+
+    /**
+     * @brief Restore a backed up secret to a vault.
+     * Restores a backed up secret, and all its versions, to a vault.
+     * This operation requires the secrets/restore permission.
+     *
+     * @param backup The backup payload as base64url encoded vector of bytes.
+     * @param context The context for the operation can be used for request cancellation.
+     *
+     * @return The Secret wrapped in the Response.
+     */
+    Azure::Response<KeyVaultSecret> RestoreSecretBackup(
```

Comment:
the Restore API should take the vector<> directly instead of the `BackupSecretData` model.

this is nit.  We can try to go different than the other languages here with this.

Reply:
That's where i'm confused here , according to your proposal backup would return a backup result , and restore would take a vector<> , which is inconsistent given the fact that these operations are the opposite of each other. 
In .Net Backup returns a byte[] and restore takes a byte[] which is consistent. returning a struct but then taking a vector<> to me seems inconsistent.

Reply:
Yes, It is all thinking about the future :) , that's what's behind the guidelines.

We can't overload a return type.  That's why we must return `Response<Model>` where Model can be updated in the future with more fields.

But, there's a little more flexibility with input parameters. We can overload methods in the future changing the input types.  So, the guideline is a little bit more flexible in this case and the only thing it mandates is that if we have `optional parameters`, we would group them all in a `operationOptions` and make it optional.  Then, any required parameters should be present.

For restore, the only required parameter right now is the std::vector<>.  So, either using the vector or the model wrapping the vector is fine because in the future we can add overloads if we need something else. 
However, if in the future `Backup` returns more things (like metadata, or something else) that is not required for `Restoring`, the restore operation will need to know that to ignore.

Also, if I, as a customer, call `Backup(...)` and I want to save the backup into a file in my system, I will only write the vector<> into a file.  Then, once I want to restore it, I can just read the file and call `restore(vector)` instead of first creating the Model. 

by the way, If you decide to keep the Model for `Restore`, make sure the model can be created by a customer by a vector<>

Reply:
based on the discussion we just had in standup i'll update it to std::vector<>

Reply:
We should always return a model because no compiled language we support can overload based on return type. However, we can overload on parameter. Our parameters are almost always simple types for required parameters, with optional parameters in optional models. That can vary, but is typically the case. Taking a model for a mere `vector<byte>` would be unnecessary.

Also, it's not a practical scenario that anyone - outside our own tests - would back up a key and immediately turn around with that same model and restore it. These are typically saved to files/blobs and later (months even years) - if necessary - restored so it would still be a byte[].